Resources/Catalog/FileCatalog.py

```diff
@@ -1,13 +1,62 @@
 """ File catalog class. This is a simple dispatcher for the file catalog plug-ins.
     It ensures that all operations are performed on the desired catalogs.
+
```

Comment:
I have basically reverted back indeed the original behavior, so no much to test actually. I think that the conclusions of our previous discussions were wrong. So, we can retry again.